src/Core/src/Graphics/MauiDrawable.Android.cs

```diff
@@ -427,18 +421,10 @@ protected override void OnDraw(Shape? shape, Canvas? canvas, APaint? paint)
 					}
 				}
 
-				if (canvas == null)
+				if (canvas == null || _clipPath == null)
 					return;
 
-				var saveCount = canvas.SaveLayer(0, 0, _width, _height, null);
-
-				if (_clipPath != null && Paint != null)
-					canvas.DrawPath(_clipPath, Paint);
-
-				if (_clipPath != null && _borderPaint != null)
-					canvas.DrawPath(_clipPath, _borderPaint);
-
-				canvas.RestoreToCount(saveCount);
+				PlatformInterop.DrawMauiDrawablePath(this, canvas, _width, _height, _clipPath, _borderPaint);
```

Comment:
If `_clipPath` is null we can return early here, no need to call `canvas.SaveLayer/RestoreToCount`.